example.tex

```diff
@@ -2,6 +2,33 @@
 \usepackage[T1]{fontenc}
 \usepackage[utf8]{inputenc}
 \usepackage{listings}
+\usepackage{xcolor}
+
+\definecolor{codegreen}{rgb}{0,0.6,0}
+\definecolor{codegray}{rgb}{0.5,0.5,0.5}
+\definecolor{codepurple}{rgb}{0.58,0,0.82}
+\definecolor{backcolour}{rgb}{0.95,0.95,0.92}
+
+\lstdefinestyle{mystyle}{
+    backgroundcolor=\color{backcolour},
+    commentstyle=\color{codegreen},
+    keywordstyle=\color{magenta},
+    numberstyle=\tiny\color{codegray},
+    stringstyle=\color{codepurple},
+    basicstyle=\ttfamily\footnotesize,
+    breakatwhitespace=false,
+    breaklines=true,
+    captionpos=b,
+    keepspaces=true,
+    numbers=left,
+    numbersep=5pt,
+    showspaces=false,
+    showstringspaces=false,
+    showtabs=false,
+    tabsize=2
+}
```

Comment:
The [package docs](https://mirrors.nic.cz/tex-archive/macros/latex/contrib/listings/listings.pdf) says:

> At the moment there are five such preferences files:
> 1. listings-acm.prf
> 2. listings-bash.prf
> 3. listings-fortran.prf
> 4. listings-lua.prf
> 5. listings-python.prf

Curiously, C is not among them.

> All contributors are invited to supply more personal preferences

Don't you want to contribute this to the listings package and only `\input` it in the example?

Reply:
C specific listings would be great, but I wonder if/when I get time. At least they have git:

https://git.gnu.org.ua/listings.git/

And C would go to https://git.gnu.org.ua/listings.git/tree/lstdrvrs.dtx

Reply:
Anyway, it's an example afterall.